File: examples/src/Text/GetTextFromSmartArtNode.cs
using System.IO

import aspose.slides as slides
using Aspose.slides.SmartArt
using System

namespace Aspose.slides.Examples.CSharp.text
{
    public class GetTextFromSmartArtNode
    {
        public static void Run()
        {
            # ExStart:GetTextFromSmartArtNode
            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_Text()

          using (Presentation presentation = new Presentation("Presentation.pptx"))
{
            slide = presentation.slides[0]
            ISmartArt smartArt = (ISmartArt)slide.shapes[0]

            ISmartArtNodeCollection smartArtNodes = smartArt.AllNodes
            foreach (ISmartArtNode smartArtNode in smartArtNodes)
            {
                foreach (ISmartArtShape nodeShape in smartArtNode.shapes)
                {
                    if (nodeShape.text_frame != None)
                        print(nodeShape.text_frame.text)
                }
            }
            }
            }
        # ExEnd:GetTextFromSmartArtNode
        }
    }

File: examples/src/text/get_text_from_smart_art_node.py
import aspose.slides as slides


def get_text_from_smart_art_node(global_opts):
    with slides.Presentation(global_opts.data_dir + "smart_art_access.pptx") as presentation:
        slide = presentation.slides[0]
        smart_art = slide.shapes[0]

        smart_art_nodes = smart_art.all_nodes
        for smart_art_node in smart_art_nodes:
            for node_shape in smart_art_node.shapes:
                if node_shape.text_frame is not None:
                    print(node_shape.text_frame.text)
